engine/server/engine/main.go

```diff
@@ -338,3 +359,87 @@ func formatFilenameFunctionForLogs(filename string, functionName string) string
 	output.WriteString("]")
 	return output.String()
 }
+
+func restApiServer(
+	ctx context.Context,
+	serverArgs *args.EngineServerArgs,
+	enclave_manager *enclave_manager.EnclaveManager,
+	perWeekLogsDatabaseClient centralized_logs.LogsDatabaseClient,
+	perFileLogsDatabaseClient centralized_logs.LogsDatabaseClient,
+	logFileManager *log_file_manager.LogFileManager,
+	metricsClient metrics_client.MetricsClient,
+) {
+	logrus.Info("Running REST API server...")
+
+	// This is how you set up a basic Echo router
+	e := echo.New()
+	// Log all requests
+	e.Use(echomiddleware.Logger())
+
+	// ============================== Engine Management API ======================================
+	// OpenAPI schema.
+	swagger_engine, err := engineApi.GetSwagger()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error loading swagger spec\n: %s", err)
+		os.Exit(1)
+	}
+	server.ServeSwaggerUI(e, "/api/specs/engine", server.NewSwaggerUIConfig(swagger_engine))
+
+	// Use our validation middleware to check all requests against the
+	// e.Use(middleware.OapiRequestValidator(swagger_engine))
+
+	// We now register our runtime above as the handler for the interface
+	engineRuntime := restApi.EngineRuntime{
+		ImageVersionTag: serverArgs.ImageVersionTag,
+		EnclaveManager:  enclave_manager,
+		LogFileManager:  logFileManager,
+		MetricsClient:   metricsClient,
+	}
+	engineApi.RegisterHandlers(e, engineApi.NewStrictHandler(engineRuntime, nil))
+
+	// ============================== Logging API ======================================
+
+	swagger_websocket, err := loggingApi.GetSwagger()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error loading swagger spec\n: %s", err)
+		os.Exit(1)
+	}
+	server.ServeSwaggerUI(e, "/api/specs/websocket", server.NewSwaggerUIConfig(swagger_websocket))
+	// Use our validation middleware to check all requests against the
+	// e.Use(middleware.OapiRequestValidator(swagger_websocket))
+	webSocketRuntime := restApi.WebSocketRuntime{
+		ImageVersionTag:             serverArgs.ImageVersionTag,
+		EnclaveManager:              enclave_manager,
+		MetricsUserID:               serverArgs.MetricsUserID,
+		DidUserAcceptSendingMetrics: serverArgs.DidUserAcceptSendingMetrics,
+		PerWeekLogsDatabaseClient:   perWeekLogsDatabaseClient,
+		PerFileLogsDatabaseClient:   perFileLogsDatabaseClient,
+		LogFileManager:              logFileManager,
+		MetricsClient:               metricsClient,
+	}
+	// TODO(edgar) add logging Close()
```

Comment:
is this TODO for the next PR's? just calling in case you forgot.

Reply:
implementation has changed but it's done